homeassistant/components/habitica/todo.py

```diff
@@ -0,0 +1,330 @@
+                translation_domain=DOMAIN,
+                translation_key=f"create_{self.idx}_item_failed",
+                translation_placeholders={"name": item.summary or ""},
+            ) from e
+
+        await self.coordinator.async_refresh()
+
+
+class HabiticaDailiesListEntity(BaseHabiticaListEntity):
+    """List of Habitica dailies."""
+
+    _attr_supported_features = (
+        TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.MOVE_TODO_ITEM
+        | TodoListEntityFeature.SET_DUE_DATE_ON_ITEM
+        | TodoListEntityFeature.SET_DESCRIPTION_ON_ITEM
+    )
+
+    def __init__(self, coordinator: HabiticaDataUpdateCoordinator) -> None:
+        """Initialize HabiticaDailiesListEntity."""
+        super().__init__(coordinator, HabiticaTodoList.DAILIES)
+
+    @property
+    def todo_items(self) -> list[TodoItem]:
+        """Return the dailies.
+
+        dailies don't have a date, but we still can show the next due date,
+        which is a calculated value based on recurrence of the task.
+        If a task is due and has not been completed, the due date is the last time
+        a new day has been started. This allows to check off dailies from yesterday,
+        that have been completed but forgotten to mark as completed before resetting the dailies.
+        Changes of the date input field in Home Assistant will be ignored.
+        """
+
+        def next_due_date(task: dict[str, Any]) -> datetime.date | None:
+            if task["isDue"] and not task["completed"]:
+                return dt_util.as_local(
+                    datetime.datetime.fromisoformat(
+                        self.coordinator.data.user["lastCron"]
+                    )
+                ).date()
+            try:
+                return dt_util.as_local(
+                    datetime.datetime.fromisoformat(task["nextDue"][0])
+                ).date()
+            except ValueError:
+                # sometimes nextDue dates are in this format instead of iso:
+                # "Mon May 06 2024 00:00:00 GMT+0200"
+                try:
+                    return dt_util.as_local(
+                        datetime.datetime.strptime(
+                            task["nextDue"][0], "%a %b %d %Y %H:%M:%S %Z%z"
+                        )
+                    ).date()
+                except ValueError:
+                    return None
+            except IndexError:
+                return None
+
+        return [
+            *(
+                TodoItem(
+                    uid=task["id"],
+                    summary=task["text"],
+                    description=task["notes"],
+                    due=next_due_date(task),
+                    status=TodoItemStatus.COMPLETED
+                    if task["completed"]
+                    else TodoItemStatus.NEEDS_ACTION,
+                )
+                for task in self.coordinator.data.tasks
+                if task["type"] == HabiticaTaskType.DAILY
```

Comment:
```suggestion
                if task["type"] is HabiticaTaskType.DAILY
```

Reply:
this seems to change the behaviour of the comparision, task["type"] is not of the type HabiticaTaskType, so we need to compare it with the value of HabiticaTaskType.DAILY. Maybe if i add proper type hints to the returned data, but that would be a to-do for another PR ^^